x-pack/plugins/observability/public/pages/alerts/alerts_table_t_grid.tsx

```diff
@@ -0,0 +1,197 @@
+import type { ActionProps, ColumnHeaderOptions, RowRenderer } from '../../../../timelines/common';
+import { getRenderCellValue } from './render_cell_value';
+import { usePluginContext } from '../../hooks/use_plugin_context';
+import { decorateResponse } from './decorate_response';
+import { LazyAlertsFlyout } from '../..';
+
+interface AlertsTableTGridProps {
+  indexName: string;
+  rangeFrom: string;
+  rangeTo: string;
+  kuery: string;
+  status: string;
+  setRefetch: (ref: () => void) => void;
+}
+
+/**
+ * columns implements a subset of `EuiDataGrid`'s `EuiDataGridColumn` interface,
+ * plus additional TGrid column properties
+ */
+export const columns: Array<
+  Pick<EuiDataGridColumn, 'display' | 'displayAsText' | 'id' | 'initialWidth'> & ColumnHeaderOptions
+> = [
+  {
+    columnHeaderType: 'not-filtered',
+    displayAsText: i18n.translate('xpack.observability.alertsTGrid.statusColumnDescription', {
+      defaultMessage: 'Status',
+    }),
+    id: ALERT_STATUS,
+    initialWidth: 79,
+  },
+  {
+    columnHeaderType: 'not-filtered',
+    displayAsText: i18n.translate('xpack.observability.alertsTGrid.triggeredColumnDescription', {
+      defaultMessage: 'Triggered',
+    }),
+    id: ALERT_START,
+    initialWidth: 116,
+  },
+  {
+    columnHeaderType: 'not-filtered',
+    displayAsText: i18n.translate('xpack.observability.alertsTGrid.durationColumnDescription', {
+      defaultMessage: 'Duration',
+    }),
+    id: ALERT_DURATION,
+    initialWidth: 116,
+  },
+  {
+    columnHeaderType: 'not-filtered',
+    displayAsText: i18n.translate('xpack.observability.alertsTGrid.severityColumnDescription', {
+      defaultMessage: 'Severity',
+    }),
+    id: ALERT_SEVERITY_LEVEL,
+    initialWidth: 102,
+  },
+  {
+    columnHeaderType: 'not-filtered',
+    displayAsText: i18n.translate('xpack.observability.alertsTGrid.reasonColumnDescription', {
+      defaultMessage: 'Reason',
+    }),
+    linkField: '*',
+    id: RULE_NAME,
+    initialWidth: 400,
+  },
+];
+
+const NO_ROW_RENDER: RowRenderer[] = [];
+
+const trailingControlColumns: never[] = [];
+
+export function AlertsTableTGrid(props: AlertsTableTGridProps) {
+  const { core, observabilityRuleTypeRegistry } = usePluginContext();
+  const { prepend } = core.http.basePath;
+  const { indexName, rangeFrom, rangeTo, kuery, status, setRefetch } = props;
+  const [flyoutAlert, setFlyoutAlert] = useState<TopAlert | undefined>(undefined);
+  const handleFlyoutClose = () => setFlyoutAlert(undefined);
+  const { timelines } = useKibana<{ timelines: TimelinesUIStart }>().services;
+
+  const leadingControlColumns = [
+    {
+      id: 'expand',
+      width: 40,
+      headerCellRender: () => null,
+      rowCellRender: ({ data }: ActionProps) => {
+        const dataFieldEs = data.reduce((acc, d) => ({ ...acc, [d.field]: d.value }), {});
+        const decoratedAlerts = decorateResponse(
+          [dataFieldEs] ?? [],
```

Comment:
nit: is the `?? []` still required here (and [here](https://github.com/elastic/kibana/pull/103270/files#diff-f9d2565b68bb75d711c69a2d1c21a6e9c4ced3494068fee60845b93e4eca1895R128))?